### ISSUE TITLE
Questions about ap50 of novel data in meta-testing

### ISSUE THREAD

**Issue by Simon-Stma** [GuangxingHan/Meta-Faster-R-CNN#11]
I would like to ask a question about the metrics of the Novel class test data in the first stage of getting three stages of training. My current understanding is that 1-1 of fewx baseline uses some base data for a basic training, while 1-2 introduces class prototypes and adds meta-RPN and meta-classifier for training, 1-3 is to solve the spatial misalignment problem, then I would like to know in 1-1, 1-2, 1-3 of these three stages of testing process, what should be the ap50 of the NOVEL class data. When I use my own dataset, the ap50 of the novel class data in all three stages is basically close to 0 (but in your paper, in Table 7, there are values of ap50 for meta-train in the base class and meta-test in the novel class, is this metric the average ap50 when testing on the novel class data?) And he really does not have the novel class data involved in the meta-training process, why meta-training can be meta-testing on the novel class with such good results?

**Comment by GuangxingHan** [comment]
Sorry for the late reply.
- We reported the meta-testing results of novel classes after the 1-3 step in the Table 7 (https://arxiv.org/pdf/2104.07719.pdf).
- The reason of the high meta-testing performance over unseen novel classes is that the query-support based detector is class-agnostic, and we sample lots of few-shot episodes from base classes during meta-training, which is similar as what we do during meta-testing. In this way, there is no surprise that our meta-trained model can be transferred to novel classes with good results.
- We also reported the meta-testing results for base-classes in Table 6, which is much better than novel classes because it is trained over base classes. This indicates that there is still much room for improvement regarding meta-testing over novel classes.

**Comment by Simon-Stma** [comment]
Thank you for your patient answer, I generally understand what you mean. I have tested the optical remote sensing datasets by different scale these days (two datasets with 800 and 23000 images respectively), and found that in the meta-training stage for meta-testing ([1,2,3,5,10] shots), the ap50 of the NOVEL class are very low, close to 0, while the other BASE class data are are more normal. This is not the same as the ap50 of VOC and COCOnovel classes given in the paper (ap50 can be 20~50 in the paper), so I wonder if this is related to the dataset? Or maybe there is some problem with the parameter configuration?

**Comment by GuangxingHan** [comment]
It seems that the learned model is overfit to base classes. 
You might need to tune some of the hyper-parameters to reduce the overfitting, like learning rate.
